src/shared/unit-file.c

```diff
@@ -217,18 +224,20 @@ bool lookup_paths_mtime_good(const LookupPaths *lp, usec_t mtime) {
                         continue;
                 }
 
-                if (timespec_load(&st.st_mtim) > mtime) {
-                        log_debug_errno(errno, "Unit dir %s has changed, need to update cache.", *dir);
-                        return false;
-                }
+                siphash24_compress_usec_t(timespec_load(&st.st_mtim), &state);
         }
 
-        return true;
+        uint64_t new = siphash24_finalize(&state);
```

Comment:
this compiles? `new()` is a macro for us...

(I guess it compiles for the only reason that there's no `(` following right after.)

Please rename this variable